core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1697,4 +1708,89 @@ public function generateRecordLabel(array $row, string $table = null, bool $prot
 
 		return $label;
 	}
+
+	/**
+	 * @param array<int|string> $ids
+	 */
+	protected function preloadCurrentRecords(array $ids, string $table): void
+	{
+		foreach ($ids as $id)
+		{
+			$this->setCurrentRecordCache($id, $table, null);
+		}
+
+		/** @var Connection $connection */
+		$connection = System::getContainer()->get('database_connection');
+
+		$stmt = $connection->executeQuery(
+			'SELECT * FROM ' . $table . ' WHERE id IN (?)',
+			array($ids),
+			array(is_numeric(array_shift($ids)) ? Connection::PARAM_INT_ARRAY : Connection::PARAM_STR_ARRAY)
+		);
+
+		foreach ($stmt->iterateAssociative() as $row)
+		{
+			if (!\is_array($row))
+			{
+				continue;
+			}
+
+			$this->setCurrentRecordCache($row['id'], $table, $row);
+		}
+	}
+
+	/**
+	 * @param array|null $row Pass null to remove a given cache entry
+	 */
+	protected function setCurrentRecordCache(string|int $id, string $table, ?array $row): void
+	{
+		if (null === $row)
+		{
+			unset($this->arrCurrentRecordCache[$table . '.' . $id]);
+
+			return;
+		}
+
+		$this->arrCurrentRecordCache[$table . '.' . $id] = $row;
+	}
+
+	public function getCurrentRecord(string|int $id = null, string $table = null, bool $noCache = false): ?array
+	{
+		$id = $id ?: $this->intId;
+		$table = $table ?: $this->strTable;
+
+		$key = $table . '.' . $id;
+
+		if ($noCache || !isset($this->arrCurrentRecordCache[$key]))
+		{
+			$this->preloadCurrentRecords(array($id), $table);
+		}
+
+		// In case this record was not part of the preloaded result, we don't need to apply any permission checks
+		if (!array_key_exists($key, $this->arrCurrentRecordCache))
```

Comment:
since we now store the exception instead of `null` we could go back to `isset` (I think is more performant?)
```suggestion
		if (!isset($key, $this->arrCurrentRecordCache))
```

Reply:
Differences are neglectable now that `array_key_exists` has its own OPCode (https://github.com/php/php-src/blob/PHP-8.1/Zend/zend_compile.c#L4416). But would need to use the global one anyway here so might as well use isset.